crates/oxc_codegen/src/lib.rs

```diff
@@ -639,13 +639,19 @@ impl<'a> Codegen<'a> {
         self.print_ascii_byte(self.quote);
     }
 
-    fn add_source_mapping(&mut self, position: u32) {
+    fn add_source_mapping(&mut self, span: Span, position: u32) {
```

Comment:
Can't you use `span.start` as `position` if all positions are `span.start`?

Reply:
There are a few cases `span.end` is explicitly emitted, for example:

https://github.com/oxc-project/oxc/blob/41d401148b773bd7256f74e7cceb95fe5576440f/crates/oxc_codegen/src/gen.rs#L815

In total, there are about 6 cases (span.end) vs. 90 cases (span.start), so I can probably add `add_source_mapping_end` dedicated for those minority. Would it be preferred?

Reply:
Yes, `add_source_mapping_end` seems better.